xDrip/Components/Charts/SupportingLogic/GlucoseChartProvider.swift

```diff
@@ -46,5 +71,175 @@ extension GlucoseChartProvider where Self: UIView {
             context.setFillColor(color.cgColor)
             context.fillEllipse(in: circleRect)
         }
+        
+        drawBasalStroke()
+    }
+    
+    func drawBasalStroke() {
+        guard let context = UIGraphicsGetCurrentContext() else { return }
+        guard let displayMode = User.current.settings.chart?.basalDisplayMode, displayMode != .notShown else { return }
+        let yIntervalBasal = yInterval / 4.0
+        let yMax = displayMode == .onBottom ? yInterval - yIntervalBasal : 0
+        let yMin = displayMode == .onBottom ? yInterval : yIntervalBasal
+        
+        guard var max = basalEntries.map({ $0.value }).max() else { return }
+        
+        let min = 0.0
+        max = max.rounded(.up)
+        
+        if max ~~ min {
+            max += 1.0
+        }
+        
+        let pxlPerVal = yIntervalBasal / (max - min)
+        
+        func calcPoint(for date: Date, and value: Double) -> CGPoint {
+            let xPoint = CGFloat((date.timeIntervalSince1970 - minDate) * pixelsPerSecond) + insets.left
+            let yPoint = CGFloat((max - value) * pxlPerVal) + insets.top + CGFloat(yMax)
+            
+            return CGPoint(x: xPoint, y: yPoint)
+        }
+        
+        var lastValue = 0.0
+        func setReducedBasalValue(prevEntry: BasalChartEntry?, currentEntry: BasalChartEntry?, in context: CGContext) {
+            guard let prevEntry = prevEntry else { return }
+            let basalRates = User.current.settings.sortedBasalRates
```

Comment:
Chart shouldn't fetch any business logic. Provide it to the chart from the outside instead.